Lib/ntpath.py

```diff
@@ -122,54 +122,169 @@ def join(path, *paths):
+
+    This is a generalization of the UNC specification in [MS-DTYP] 2.2.57. The
+    latter specifies the file namespace, for which the domain is referred to
+    as "host-name" (more generally "server") and the junction as "share-name".
+    The server is commonly a local or remote network name (i.e. NETBIOS name,
+    DNS name, or IP address). It can also be a non-network server provided by
+    a local redirector. The share is a resource provided by the server, such
+    as a file-system directory.
+
+    UNC drive examples in the file namespace:
+
+        splitdrive('//server/share') == ('//server/share', '')
+        splitdrive('//server///share') == ('//server///share', '')
+        splitdrive('//server/share/') == ('//server/share', '/')
+        splitdrive('//server/share/dir') == ('//server/share', '/dir')
+
+    The other supported namespace is the device namespace, which is mapped as
+    two domains, "." and "?". These domains are handled differently in some
+    contexts, such as when creating or opening a file, but for our puposes
+    here they are equivalent. In this namespace, the junction is case-
+    insensitive. Any device junction is recognized as a UNC drive, with
+    two exceptions that require additional qualification: "GLOBAL" and "UNC".
+
+    Normally the device namespace includes the local device junctions of a
+    user, such as mapped and subst drives. The "GLOBAL" junction limits this
+    view to just global devices. It must be followed either by a device
+    junction or another "GLOBAL" junction.
+
+    The equivalent of the UNC file namespace in the device namespace is the
+    "UNC" device junction, but only when there is a remaining path (e.g. at
+    least a trailing separator). For consistency with the file namespace, if
+    the "UNC" device junction has a reminaing path, it must include a server
+    and share in order to be recognized as a drive.
+
+    UNC drive examples in the device namespace:
+
+        splitdrive('//./C:') == ('//./C:', '')
+        splitdrive('//?/C:/dir') == ('//?/C:', '/dir')
+
+        splitdrive('//./UNC') == ('//./UNC', '')
+        splitdrive('//?/UNC/server/share') == ('//?/UNC/server/share', '')
+        splitdrive('//?/UNC/server/share/dir') == (
+            '//?/UNC/server/share', '/dir')
+
+        splitdrive('//./Global/C:') == ('//./Global/C:', '')
+        splitdrive('//?/Global/Global/C:/') == ('//?/Global/Global/C:', '/')
+        splitdrive('//?/Global/UNC/server/share/dir') == (
+            '//?/Global/UNC/server/share', '/dir')
+
+    Examples with no drive:
+
+        splitdrive('') == ('', '')
+        splitdrive('dir') == ('', 'dir')
+        splitdrive('/dir') == ('', '/dir')
+
+        splitdrive('//') == ('', '//')
+        splitdrive('//server/') == ('', '//server/')
+        splitdrive('///server/share') == ('', '///server/share')
+
+        splitdrive('//?/UNC/') == ('', '//?/UNC/')
+        splitdrive('//?/UNC/server/') == ('', '//?/UNC/server/')
```

Comment:
As mentioned above, to align with `PathCchSkipRoot()`, these examples should have a drive, as follows:

```
 splitdrive('//') == ('//', '')
 splitdrive('//server/') == ('//server/', '')
 splitdrive('///share/file') == ('///share', '/file')
 splitdrive('//?/UNC/') == ('//?/UNC/', '')
 splitdrive('//?/UNC/server/') == ('//?/UNC/server/', '')
```
I changed some component names to reflect how they're classified. These examples can be included in a section about paths with malformed drives.